File: modules/mira-face-detector/ios/MiraFaceDetectorModule.swift
import ExpoModulesCore
import VisionCamera

public class MiraFaceDetectorModule: Module {
  public func definition() -> ModuleDefinition {
    Name("MiraFaceDetector")

    OnCreate {
      VisionCameraProxyHolder.setFrameProcessorPlugin("miraFaceDetector") { proxy, options in
        return MiraFaceDetectorPlugin(proxy: proxy, options: options)
      }
    }
  }
}

File: modules/mira-face-detector/android/src/main/java/expo/modules/mirafacedetector/MiraFaceDetectorModule.kt
package expo.modules.mirafacedetector

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import com.mrousavy.camera.frameprocessors.FrameProcessorPluginRegistry

class MiraFaceDetectorModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("MiraFaceDetector")

    OnCreate {
      FrameProcessorPluginRegistry.addFrameProcessorPlugin("miraFaceDetector") { proxy, options ->
        MiraFaceDetectorPlugin(proxy, options)
      }
    }
  }
}
